### DISCUSSION TITLE
Opentelemetry Broadway

### DISCUSSION THREAD

**Discussion by tielur** [open-telemetry/opentelemetry-erlang-contrib#97] [Ideas]
I've been looking into adding open telemetry into a few projects at work. I found `OpentelemetryPhoenix` which I thought was neat how it used the [telemetry](https://hexdocs.pm/telemetry/) handlers to create `OpenTelemetry` spans. I'm wondering if the project is open to adding a Broadway instrumentation that would work in a similar fashion to hook into [broadway telemetry events](https://hexdocs.pm/broadway/Broadway.html#module-telemetry) to create spans?

If so, I'd love to take a crack at it

**Comment by tsloughter** [comment]
Definitely!

Does Broadway provide an interface on top of writing as well as reading? My first thought was it'd be great if Broadway made it so a common propagation interface of trace context in for example kafka headers would be possible.

**Reply by andrewhr** [reply]
I got you back, @tielur :) https://github.com/open-telemetry/opentelemetry-erlang-contrib/pull/92

About propagation, I did some experiments at work (and running them in prod). I'm just unure the approach we did would be vetted by Broadway team or is too much of a hack. I the PR with current implementation gets traction, maybe we can involve them.

**Reply by andrewhr** [reply]
@tsloughter alternatively, we could proceed with the PR and publish a 0.1.0 version or something. I you're ok with that approach I can do the required updates and get ready to merge mainstream.

**Reply by tielur** [reply]
@andrewhr Any movement on this? Would love to get broadway tracing. Are we just waiting for upstream approval?

**Reply by medoror-mo** [reply]
this seems promising: https://github.com/breakroom/opentelemetry_broadway
